odoo_regex_set/task.py

```diff
@@ -0,0 +1,33 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from typing import List
+import re
+
+task_details_from_tag_regex = re.compile(
+    r"(^|\s+)(?P<task_tag>[Tt][Aa]#?(?P<id>\d+))(\s+|$)"
+)
+
+
+def find_task_details(text: str) -> (str, str):
+    no_parentheses = re.sub(r"([()])", " ", text)
+    res = re.search(task_details_from_tag_regex, no_parentheses)
+
+    if res:
+        group_dict = res.groupdict()
+        return group_dict["id"], group_dict["task_tag"]
+
+    return "", ""
+
+
+def remove_task_from_text(text: str, task_tag: str) -> str:
+    return re.sub(task_tag, "", text)
+
+
+def find_ids(text: str, ids=None) -> List[str]:
+    ids = ids or set([])
+    id_, task_tag = find_task_details(text)
+    if id_:
+        ids.add(id_)
+        cleaned_text = remove_task_from_text(text, task_tag)
+        ids = find_ids(cleaned_text, ids)
```

Comment:
@foutoucour selon clean code, la fonction appelée devrait être définie plus bas que la fonction appelant. Page 37 du livre (The Stepdown Rule).

Reply:
Ça serait bien de rendre find_task_details et remove_task_from_text privées. Le but est de rendre explicite qu'on s'attend à ce que ces fonctions ne soient pas appelées en dehors de la librairie.

Reply:
J'ai ré-ordonné les functions et j'ai mis les tools dans un fichier séparé pour aider la compréhension.
Il n'y a pas de façon de rendre privée une fonction dans python au final. Il faut faire confiance :)
```
In Python, "privacy" depends on "consenting adults'" levels of agreement - you can't force it (any more than you can in real life;-)
```
https://stackoverflow.com/questions/1547145/defining-private-module-functions-in-python

Reply:
@foutoucour le but n'est absolument pas de forcer une fonction privée.

Le préfixe `_` est une convention très connue. Ça rend le code beaucoup plus lisible.